src/components/PreferredCurrencySelector.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogContent,
+  DialogTitle,
+  Grid,
+  IconButton,
+  Stack,
+  styled,
+  Typography
+} from '@mui/material';
+import { useCallback, useState } from 'react';
+import { usePreferredCurrencies } from '../hooks/usePreferredCurrencies';
+import { useAppDispatch, useAppState } from '../store';
+import { emptyFunction } from '../utils/common';
+import { CurrencyCode } from '../utils/currencies';
+import Iconify from './Iconify';
+
+interface StyledStackProps {
+  selected: boolean;
+}
+
+const StyledStack = styled(Stack, {
+  shouldForwardProp: (prop) => prop !== 'selected'
+})<StyledStackProps>(({ theme, selected }) => ({
+  padding: `${theme.spacing(1)} ${theme.spacing(2.5)}`,
+  border: selected ? `1px solid ${theme.palette.grey[700]}` : '1px solid transparent',
+  cursor: 'pointer',
+  '&:hover': {
+    border: `1px solid ${theme.palette.grey[300]}`
+  }
+}));
+
+const CurrencyItem = ({ code, name, selected = false, onClick = emptyFunction }) => {
+  return (
+    <StyledStack selected={selected} onClick={onClick}>
+      <Typography>{name}</Typography>
+      <Typography>{code}</Typography>
+    </StyledStack>
+  );
+};
+
+export const PreferredCurrencySelector = () => {
+  const { account, userSettings } = useAppState();
+  const dispatch = useAppDispatch();
+  const { preferredCurrencies } = usePreferredCurrencies();
+  const [dialogOpen, setDialogOpen] = useState(false);
+
+  const openDialog = () => setDialogOpen(true);
+  const closeDialog = () => setDialogOpen(false);
+
+  const handleCurrencyChange = useCallback(
```

Comment:
we discussed the option to put the dispatch in the `userUserSettings` hook, i know see that this hook is empt. Could you eleborate?